### ISSUE TITLE
`podman build` appears to ignore `--arch` option when pulling multiarch images

### ISSUE THREAD

**Issue by srcshelton** [containers/podman#8894]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I'm trying to build a non-native container with `podman build --arch arm64 Dockerfile` on an armv7a userland.  I don't necessarily expect this to work.

I would, however, hope that `podman` would pull the `arm64` variant of the multiarch image referenced (by tag ("latest"), rather than by sha256 hash) in the `Dockerfile` file since the `--arch` option was given.  This may not actually be usable once fetched - but the override should, I believe, still apply since it was explicitly given.

(Re-reading the help text suggests that the value of `--arch` might only be used when adding parameters to a newly created image - in which case please consider this instead a Feature Request to either add an option to override the architecture pulled, or dual-purpose this existing option!)

**Steps to reproduce the issue:**

1. Pull a multiarch image on an `arm` host with the `--arch` argument to `podman build` set to `arm64`

2. Inspect the pulled image.

**Describe the results you received:**

```
"Architecture": "arm"
```

**Describe the results you expected:**

```
"Architecture": "arm64"
```

**Output of `podman version`:**

```
Version:      2.2.1
API Version:  2.1.0
Go Version:   go1.15.5
Git Commit:   a0d478edea7f775b7ce32f8eb1a01e75374486cb
Built:        Thu Dec 31 13:48:19 2020
OS/Arch:      linux/arm
```

**Output of `podman info --debug`:**

```
host:
  arch: arm
  buildahVersion: 1.18.0
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 2.0.22, commit: 9c34a8663b85e479e0c083801e89a2b2835228ed'
  cpus: 4
  distribution:
    distribution: gentoo
    version: unknown
  eventLogger: file
  hostname: turnpike
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.4.79-v8+
  linkmode: dynamic
  memFree: 3359043584
  memTotal: 8194351104
  ociRuntime:
    name: crun
    package: Unknown
    path: /usr/bin/crun
    version: |-
      crun version 0.16
      commit: eb0145e5ad4d8207e84a327248af76663d4e50dd
      spec: 1.0.0
      +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 129h 34m 25.56s (Approximately 5.38 days)
registries:
  localhost:5000:
    Blocked: false
    Insecure: true
    Location: localhost:5000
    MirrorByDigestOnly: false
    Mirrors: []
    Prefix: localhost:5000
  search:
  - docker.io
  - docker.pkg.github.com
  - quay.io
  - public.ecr.aws
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions:
    overlay.ignore_chown_errors: "false"
  graphRoot: /storage/containers/podman/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 26
  runRoot: /storage/containers/podman/run
  volumePath: /storage/containers/podman/volumes
version:
  APIVersion: 2.1.0
  Built: 1609422499
  BuiltTime: Thu Dec 31 13:48:19 2020
  GitCommit: a0d478edea7f775b7ce32f8eb1a01e75374486cb
  GoVersion: go1.15.5
  OsArch: linux/arm
  Version: 2.2.1

```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Comment by rhatdan** [comment]
Their are hidden options
--override-arch and --override-variant that will allow you to move forward.  I am working on fixing this in buildah, now and will move to fixing it in Podman as soon as the buildah merges.

https://github.com/containers/buildah/pull/2868

**Comment by rhatdan** [comment]
Fixed in podman 3.0
```
$ arch
x86_64
$ ./bin/podman run --arch arm64 ubi8 arch
Resolved "ubi8" as an alias (/etc/containers/registries.conf.d/shortnames.conf)
Trying to pull registry.access.redhat.com/ubi8:latest...
Getting image source signatures
Checking if image destination supports signatures
Copying blob cbe902a0a8c4 skipped: already exists  
Copying blob e753ad39f085 [--------------------------------------] 0.0b / 0.0b
Copying config 70dab2c4ec done  
Writing manifest to image destination
Storing signatures
aarch64
$ ./bin/podman build --arch arm64 /tmp/
STEP 1: FROM ubi8
STEP 2: RUN arch
aarch64
--> Using cache e6de2269f2fd6d4a7e9edb93aa6862c22e5d4ed3053445ed2e3e3a528fb30f6e
--> e6de2269f2f
e6de2269f2fd6d4a7e9edb93aa6862c22e5d4ed3053445ed2e3e3a528fb30f6e
```
